test/ci/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+include(TileDBAssertions)
+
+find_package(Catch_EP REQUIRED)
+
+add_executable(
+  test_assert
+  test_assert.cc
+)
+
+target_link_libraries(test_assert PUBLIC Catch2::Catch2)
+
+if (TILEDB_ASSERTIONS)
+  target_compile_definitions(
+    test_assert
+    PRIVATE
+    -DTILEDB_ASSERTIONS
```

Comment:
Maybe "-DTILEDB_ASSERTIONS=1" for somewhat more consistency with other uses of similar/like names in this effort,
to save others like me the trouble of figuring out whether definition value matters in this context?

Reply:
Keeping as-is for consistency with other existing compile definitions (vs cmake options).